File: signal-alert/modules/telegram.py
"""텔레그램 알림."""

import json
import logging
import os
import urllib.parse
import urllib.request

logger = logging.getLogger("signal-alert")


def send_telegram(message: str) -> bool:
    """텔레그램 봇으로 평문 메시지 전송."""
    bot_token = os.environ.get("TELEGRAM_BOT_TOKEN", "")
    chat_id = os.environ.get("TELEGRAM_CHAT_ID", "")

    if not bot_token or not chat_id:
        logger.warning("텔레그램 설정 없음")
        return False

    url = f"https://api.telegram.org/bot{bot_token}/sendMessage"
    data = urllib.parse.urlencode({
        "chat_id": chat_id,
        "text": message,
    }).encode("utf-8")

    try:
        req = urllib.request.Request(url, data=data)
        with urllib.request.urlopen(req, timeout=10) as resp:
            result = json.loads(resp.read())
            if result.get("ok"):
                logger.info("텔레그램 전송 성공")
                return True
            logger.warning("텔레그램 응답 오류: %s", result)
            return False
    except Exception as e:
        logger.error("텔레그램 전송 실패: %s", e)
        return False

File: ddsaopal/src/telegram.ts
// Alram🔔 봇 발신 (plain text, 마크다운 미사용). 실패해도 흐름 방해 안 함.
import https from "node:https";
import type { Env } from "./tossApi.js";

export async function notify(env: Env, text: string): Promise<void> {
  if (!env.TELEGRAM_BOT_TOKEN || !env.TELEGRAM_CHAT_ID) {
    console.log("[telegram 미설정, 스킵]\n" + text);
    return;
  }
  const body = JSON.stringify({ chat_id: env.TELEGRAM_CHAT_ID, text });
  await new Promise<void>((resolve) => {
    const req = https.request(
      {
        hostname: "api.telegram.org",
        path: `/bot${env.TELEGRAM_BOT_TOKEN}/sendMessage`,
        method: "POST",
        headers: { "Content-Type": "application/json", "Content-Length": Buffer.byteLength(body) },
      },
      (res) => { res.on("data", () => {}); res.on("end", () => resolve()); },
    );
    req.on("error", (e) => { console.error("telegram 실패:", e.message); resolve(); });
    req.write(body);
    req.end();
  });
}
